app/templates/account/settings.hbs

```diff
@@ -76,4 +76,91 @@
       {{loading-indicator}}
     {{/if}}
   </section>
+  {{#if features.proVersion}}
+    <section class="active-visiblity-settings" data-test-active-visiblity-settings>
+      <h3>Insights Visibility Settings</h3>
+      <div class="insights-settings">
+        {{!-- <div class="name" data-test-insights-settings-title>Insights</div> --}}
+        <p class="description" data-test-insights-settings-description>
+          Make more informed decisions about your development workflow using your build Insights.<br />
+          {{#link-to 'owner.repositories' account.login (query-params tab='insights')}}View {{account.name}}'s Insights{{/link-to}}
+        </p>
+
+        <ul class='travis-radio-list'>
+          <li class='travis-radio-list__item'>
+            <label for="user_insight_private">Do not allow everyone to see insights from your private builds</label>
+            <input
+              type="radio"
+              name="user_insight_vis"
+              class="travis-radio-input"
+              id="user_insight_private"
+              value="private"
+              checked={{eq privateInsightsVisibility 'private'}}
+              onchange={{action (mut privateInsightsVisibility) 'private'}}
+            />
+            <label for="user_insight_private" class="travis-radio-label"></label>
+          </li>
+          <li class='travis-radio-list__item'>
+            <label for="user_insight_public">Allow everyone to see insights from your private builds</label>
+            <input
+              type="radio"
+              name="user_insight_vis"
+              class="travis-radio-input"
+              id="user_insight_public"
+              value="public"
+              checked={{eq privateInsightsVisibility 'public'}}
+              onchange={{action (mut privateInsightsVisibility) 'public'}}
+            />
+            <label for="user_insight_public" class="travis-radio-label"></label>
+          </li>
+          <li class='travis-radio-list__item travis-radio-list__item--save'>
+            <button class="settings-button" {{action 'toggleInsightsVisibilityModal'}} disabled={{eq visibilityChange 0}}>Save</button>
+          </li>
+        </ul>
+      </div>
+    </section>
+  {{/if}}
 </section>
+
+{{#if (and features.proVersion isShowingInsightsVisibilityModal)}}
+  {{#modal-dialog
+    toggleInsightsVisibilityModal=(action 'toggleInsightsVisibilityModal')
```

Comment:
hmm now that I see you’re passing this action to another component, my commentary about mutating in the template makes less sense… it could perhaps be done with `let` to declare the action in the template, but I’m not sure that’s too much of an improvement

I tried this out and had an accessibility observation: I think the keyboard focus should be moved to the first button when the modal is shown. It seems [currently untenable](https://github.com/yapplabs/ember-modal-dialog/issues/236) to remove the possibility of focus on elements behind the modal, but I think it should be possible to focus the button so at least focus is in the correct general area?

Reply:
I think if I switch the actions to all use the toggle helper, I might be able to avoid needing to pass the action in. Or maybe at least pass in the toggle helper version of the action? I'll try some things out and see how it goes.

And I definitely will try to see what I can do about keyboard focus on the modal. That's an excellent point, thanks!